README.md

```diff
@@ -56,186 +50,318 @@ npm install @dojo/has
 
 ```ts
-w<W extends WidgetBaseInterface>(widgetConstructor: W | RegistryLabel, properties: W['properties']): WNode<W>;
-```
+interface ListItemProperties extends WidgetProperties {
+	id: string;
+	content: string;
+	highlighted: boolean;
+	onItemClick: (id: string) => void;
+}
 
-The following code creates a widget using the `widgetConstructor`, `properties` and `children`
+class ListItem extends WidgetBase<ListItemProperties> {
 
-```ts
-w<W extends WidgetBaseInterface>(widgetConstructor: W | RegistryLabel, properties: W['properties'], children: W['children']): WNode<W>;
+	protected onClick(event: MouseEvent) {
+		const { id } = this.properties;
+
+		this.properties.onItemClick(id);
+	}
+
+	protected render(): DNode {
+		const { id, content, highlighted } = this.properties;
+		const classes = highlighted ? { highlighted: true } : { highlighted: false };
+
+		return v('li', { key: id, classes, onclick: this.onClick }, [ content ]);
+	}
+}
 ```
-Example `w` constructs:
+
+Using the `ListItem` we can simplify the `List` slightly and also add the on click functionality that we required:
 
 ```ts
-w(WidgetClass, properties);
-w(WidgetClass, properties, children);
+interface ListProperties extends WidgetProperties {
+	items: {
+		id: string;
+		content: string;
+		highlighted: boolean;
+	};
+	onItemClick: (id: string) => void;
+}
+
+class List extends WidgetBase<ListProperties> {
+	protected render() {
+		const { onItemClick, items } = this.properties;
 
-w('my-widget', properties);
-w('my-widget', properties, children);
+		return v('ul', { classes: { list: true } }, items.map(({ id, content, highlighted }) => {
+			return w(ListItem, { key:id, content, highlighted, onItemClick });
+		});
+	}
+}
 ```
 
-The example above that uses a string for the `widgetConstructor `, is taking advantage of the [widget registry](#widget-registry) functionality.
-The widget registry allows for the lazy loading of widgets.
+Additionally the `ListItem` is now reusable in other areas of our application(s).
 
-### tsx
+### Mixins
 
-In additional to the programatic functions `v` and `w`, widget-core optionally supports the use of the `jsx` syntax known as [`tsx`](https://www.typescriptlang.org/docs/handbook/jsx.html) in TypeScript.
+Dojo 2 makes use of mixins to decorate additional functionality and properties to existing widgets. Mixins provide a mechanism that allows loosely coupled design and composition of behaviors into existing widgets without having to change the base widget.
 
-To start to use `jsx` in your project the widgets need to be named with a `.tsx` extension and some configuration is required in the project's `tsconfig.json`:
+TypeScript supports mixins using a constructor type of `new (...args: any[]) => any;` that enables a class to be passed as a function argument and extended to add new functionality.
 
-Add the configuration options for `jsx`:
+Example mixin that adds method `setState` and `readonly` `state` property:
```

Comment:
Do we really want to show a mixin we consider anti-pattern?  I can't think of another good example ATM though.